pkg/provider/loadBalancer.go

```diff
@@ -238,45 +287,96 @@ func syncLoadBalancer(ctx context.Context, kubeClient kubernetes.Interface, serv
 	return &service.Status.LoadBalancer, nil
 }
 
-func discoverPool(cm *v1.ConfigMap, namespace, configMapName string) (pool string, global bool, err error) {
-	var cidr, ipRange string
+func getConfigWithNamespace(cm *v1.ConfigMap, namespace, name string) (value, key string, err error) {
 	var ok bool
 
-	// Find Cidr
-	cidrKey := fmt.Sprintf("cidr-%s", namespace)
-	// Lookup current namespace
-	if cidr, ok = cm.Data[cidrKey]; !ok {
-		klog.Info(fmt.Errorf("no cidr config for namespace [%s] exists in key [%s] configmap [%s]", namespace, cidrKey, configMapName))
-		// Lookup global cidr configmap data
-		if cidr, ok = cm.Data["cidr-global"]; !ok {
-			klog.Info(fmt.Errorf("no global cidr config exists [cidr-global]"))
+	key = fmt.Sprintf("%s-%s", name, namespace)
+
+	if value, ok = cm.Data[key]; !ok {
+		return "", key, fmt.Errorf("no config for %s", name)
+	}
+
+	return value, key, nil
+}
+
+func getConfig(cm *v1.ConfigMap, namespace, configMapName, name, configType string) (value string, global bool, err error) {
```

Comment:
```suggestion
func getConfig(cm *v1.ConfigMap, namespace, configMapName, name, configType string) (value string, global bool, found bool) {
```

Since it's just found or not found, no need to return error here